packages/native-federation-typescript/src/lib/typeScriptCompiler.ts

```diff
@@ -42,10 +41,15 @@ const createHost = (mapComponentsToExpose: Record<string, string>, tsConfig: typ
   return host
 }
 
+const createVueTscProgram = (programOptions: typescript.CreateProgramOptions) => {
+  const vueTypescript = require('vue-tsc')
```

Comment:
add a try..catch block since vue-tsc is optional.

Reply:
It is required (and imported) only if the `compilerInstance` option is set to `vue-tsc`.
IMO it should crash if you are requiring it without having it installed.